specs/SharedBuffer.md

```diff
@@ -0,0 +1,282 @@
+  /// 
+  HRESULT ShareBufferToScript(
+    [in] ICoreWebView2StagingSharedBuffer* sharedBuffer,
+    [in] BOOL isReadOnlyToScript,
+    [in] LPCWSTR additionalDataAsJson);
+}
+
+interface ICoreWebView2StagingFrame2 : IUnknown {
+  /// Share a shared buffer object with script of the iframe in the WebView.
+  /// The script will receive a `SharedBufferReceived` event from chrome.webview.
+  /// The event arg for that event will have the following properties:
+  ///   `sharedBuffer`: an ArrayBuffer object with the backing content from the shared buffer.
+  ///   `data`: an object representing `additionalDataAsJson`. This property will be `undefined`
+  ///           if `additionalDataAsJson` is nullptr or empty string.
+  ///   `source`: with a value set as `chrome.webview` object.
+  /// If `isReadOnlyToScript` is true, the script will only have read access to the buffer.
+  /// If the script tries to modify the content in a read only buffer, it will cause an access violation
+  /// in WebView renderer process and crash the renderer process.
+  /// If the shared buffer is already closed, the API will fail with HRESULT_FROM_WIN32(ERROR_INVALID_STATE).
+  /// 
+  /// The script code should call `chrome.webview.releaseSharedBuffer` with
+  /// the shared buffer as the parameter to release underlying resources as soon
+  /// as it does not need access the shared buffer any more.
+  ///
+  /// Sharing a buffer to script has security risk. You should only share buffer with trusted site.
+  /// 
+  HRESULT ShareBufferToScript(
+    [in] ICoreWebView2StagingSharedBuffer* sharedBuffer,
+    [in] BOOL isReadOnlyToScript,
+    [in] LPCWSTR additionalDataAsJson);
+}
+
+```
+
+## .NET
+```c#
+namespace Microsoft.Web.WebView2.Core
+{
+
+    class CoreWebView2Environment
+    {
+        public CoreWebView2SharedBuffer CreateSharedBuffer(ulong Size);
+    }
+
+    class CoreWebView2SharedBuffer
+    {
+        public ulong Size { get; };
+        public IntPtr Buffer { get; };
+        public IntPtr Handle { get; };
+        public Stream GetAccessStream();
+        void Close();
+    }
+    
+    runtimeclass CoreWebView2
+    {
+        public void ShareBufferToScript(CoreWebView2SharedBuffer sharedBuffer, bool isReadOnlyToScript, string additionalDataAsJson);
+    }
+    
+    class CoreWebView2Frame
+    {
+        public void ShareBufferToScript(CoreWebView2SharedBuffer sharedBuffer, bool isReadOnlyToScript, string additionalDataAsJson);
+    }
+}
+
+```
+## WinRT
+```c#
+namespace Microsoft.Web.WebView2.Core
+{
+
+    runtimeclass CoreWebView2Environment
+    {
+        [interface_name("Microsoft.Web.WebView2.Core.ICoreWebView2StagingEnvironment2")]
+        {
+            CoreWebView2SharedBuffer CreateSharedBuffer(UInt64 Size);
+        }
+    }
+
+    runtimeclass CoreWebView2SharedBuffer
+    {
+        UInt64 Size { get; };
+        Windows.Storage.Streams.IRandomAccessStream GetAccessStream();
```

Comment:
Buffer is a property as it is basically the address of the buffer and never changes.
The access stream is a separate object that is created every time for accessing the memory as stream. This is similar to [MemoryMappedFile.CreateViewStream Method](https://docs.microsoft.com/en-us/dotnet/api/system.io.memorymappedfiles.memorymappedfile.createviewstream?view=net-6.0).

Reply:
A method instead of a property makes sense, because we need to get different objects each time you call the object since each stream has its own state of where it is in the buffer.

For the name maybe `GetRandomAccessStream` or `GetStream`. I'm not sure what just the word 'Access' on its own is adding.

Reply:
Renamed as GetStream.